File: Problem1.java
// Time Complexity : O(n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No
class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0;
        int mini = prices[0];
        for(int i = 1; i < prices.length; i++){
            profit += Math.max(0,prices[i]-prices[i-1]);
        }
        return profit;
    }
}

File: Problem1.cs
public class Solution
	{
        // Time Complexity : O(n)
        // Space Complexity : O(1) - in place
        // Did this code successfully run on Leetcode : Yes
        // Any problem you faced while coding this : No
        public int MaxProfit(int[] prices)
        {
            if (prices == null || prices.Length == 0) return 0;
            int maxProfit = 0;

            int n = prices.Length;
            for (int i = 1; i < n; i++)
            {
                if (prices[i] > prices[i - 1])
                {
                    maxProfit = maxProfit + (prices[i] - prices[i - 1]);
                }
            }
            return maxProfit;
        }
    }
